### DISCUSSION TITLE
Dexcom Bridge not working as uploader to newly created Google Cloud Nightscout

### DISCUSSION THREAD

**Discussion by Erdella** [NightscoutFoundation/xDrip#2835] [Q&A]
I cannot, for the life of me, get the Dexcom bridge working on my son's new NS. I have it set up on Google Cloud. I have tried re-creating the dependent account, set up my wife and I as followers, but still no go. We have a SugarPixel, which works fine with the Dexcom credentials. Anyone else have issues with using Dexcom bridge as an uploader? Anyone know of any logs I can look at to see what's going on?

I am using my own private DNS, but it also doesn't work with FreeDNS as that is currently set up as well.

![image](https://github.com/NightscoutFoundation/xDrip/assets/53439889/7ed62986-26cf-41ae-8ceb-5726aea0bc88)

**Comment by Navid200** [comment]
If you enter the hostname in a browser on the phone that your uploader runs on, do you see your Nightscout?

If no, you need to resolve that first.  See if there are any settings on the phone that may be blocking.

If yes, you can try changing the AUTH_DEFAULT_ROLES to see if it makes a difference.
I have explained the details (and possible consequence) here:  https://navid200.github.io/xDrip/docs/Nightscout/Auth_Default_Roles.html

**Comment by Erdella** [comment]
Looks like the same exact issue you commented on the CB group here:
https://www.facebook.com/groups/cgminthecloud/permalink/6230801380374399/?mibextid=W9rl1R.

 I suspect that if I change it to readable it will work. I will confirm later. 

Is this expected behavior? If we change it to “readable”, doesn’t that mean it won’t use tokens? I’ve read that article, and understand it. Personally I’d prefer to leave it locked down if possible. 

Seems like a bug, if it’s expected to work by setting the appropriate variables and leaving the AUTH_DEFAULT_ROLES to not readable.

**Reply by Navid200** [reply]
If an app cannot communicate with Nightscout using either API_SECRET or a token, I consider that to be a bug of that app, yes.
